tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py

```diff
@@ -6,13 +6,30 @@
 from importlib.resources import files
 import json
 
+from .constants import CONNECTOR_BUILD_OUTPUT_URL
+
 from google.oauth2 import service_account
 import requests
 import pandas as pd
+from typing import Optional
+
+def get_connector_build_output_url(connector_technical_name: str, connector_version: str) -> str:
+    return f"{CONNECTOR_BUILD_OUTPUT_URL}/{connector_technical_name}/{connector_version}.json"
+
+def fetch_latest_build_status_for_connector_version(connector_technical_name: str, connector_version: str) -> Optional[str]:
+    """Fetch the latest build status for a given connector version."""
+    connector_build_output_url = get_connector_build_output_url(connector_technical_name, connector_version)
+    connector_build_output_response = requests.get(connector_build_output_url)
 
+    # if the connector returned successfully, return the outcome
+    if connector_build_output_response.status_code == 200:
+        connector_build_output = connector_build_output_response.json()
+        return connector_build_output["outcome"]
```

Comment:
nit: From the test below, this would be the string "success" if all went well. Maybe this method should return a Boolean?  Or, perhaps be an ENUM type  for the returned string values?

Reply:
I like this idea.

Originally it was a bit awkward as the build outcome being a string was already used in a lot of placed.

I wanted to go and update that as well but it felt a bit out of scope.

So I opted in [validation.py](https://github.com/airbytehq/airbyte/pull/22029/files#diff-fdd65838aa29c584c0ea92de3e6b169a0e054d356828982b37f87c1ec804e501R39) to transform the string to a boolean.

Putting at least an enum in input and using it in validation is a good idea though. I will change.